webapp/src/app/standing-table/standing-table.component.ts

```diff
@@ -16,183 +16,41 @@ export class StandingTableComponent implements OnChanges {
     @Input() noFilterSelected : Boolean;
 
     statisticsList : Array<Statistics>;
-    matchesList : Array<Match>;
     days = 0;
     numberOfMatches = 0;
     limit = 0;
-    players : Array<Player>;
 
     constructor(private championshipService : ChampionshipService, private loginService : LoginService) { }
 
     ngOnChanges(changes : SimpleChanges) {
         this.setLimit();
-        this.players = this.getPlayersFromMatches();
 
-        this.statisticsList = [];
-        this.matchesList = [];
-
-        this.createStatistics();
+        this.statisticsList = Statistics.getStatisticsFromMatches(this.matches, this.limit);
 
         // we should not update the stats it the statisticsList
         // contains stats for all championships
         if (!changes['noFilterSelected']) {
-            this.updateSeasonData();
-        }
-    }
-
-    getPlayersFromMatches() : Array<Player> {
-        const players = [];
-        for (const match in this.matches) {
-            if (match) {
-                if (players.indexOf(this.matches[match].player1) < 0) {
-                    players.push(this.matches[match].player1);
-                }
-                if (players.indexOf(this.matches[match].player2) < 0) {
-                    players.push(this.matches[match].player2);
-                }
-                if (players.indexOf(this.matches[match].player3) < 0) {
-                    players.push(this.matches[match].player3);
-                }
-                if (players.indexOf(this.matches[match].player4) < 0) {
-                    players.push(this.matches[match].player4);
-                }
-            }
+            this.updateStats();
         }
-        return players;
     }
 
-    updateSeasonData() {
-        // update the finalized season data only if the user is an admin
+    updateStats() {
+        // update the stats only if the user is an admin
         if (this.loginService.isLoggedIn()) {
             if (this.statisticsList.length > 0) {
-                const season = this.championshipService.getSelectedChampionship();
-                // we should not add stats for an unfinished season
```

Comment:
That commentary doesn't make sense to me, care to explain what happens here?

Reply:
it was already there, I didn't change it

Reply:
The stats referred here are the ones that are shown on the HoF.

Reply:
That's very confusing, care to change later on @kiosia ? Just to differentiate stats of the championship and stats of hof.